ZuluContent/Engines/Craft/DefCarpentry.cs

```diff
@@ -1,255 +1,331 @@
-			AddSkill( index, SkillName.Tailoring, 65.0, 70.0 );
-			AddRes( index, typeof( Cloth ), 1044286, 25, 1044287 );
-			index = AddCraft( typeof( SpinningwheelSouthDeed ), 1044298, 1044342, 73.6, 98.6, typeof( Log ), 1044041, 75, 1044351 );
-			AddSkill( index, SkillName.Tailoring, 65.0, 70.0 );
-			AddRes( index, typeof( Cloth ), 1044286, 25, 1044287 );
-			index = AddCraft( typeof( LoomEastDeed ), 1044298, 1044343, 84.2, 109.2, typeof( Log ), 1044041, 85, 1044351 );
-			AddSkill( index, SkillName.Tailoring, 65.0, 70.0 );
-			AddRes( index, typeof( Cloth ), 1044286, 25, 1044287 );
-			index = AddCraft( typeof( LoomSouthDeed ), 1044298, 1044344, 84.2, 109.2, typeof( Log ), 1044041, 85, 1044351 );
-			AddSkill( index, SkillName.Tailoring, 65.0, 70.0 );
-			AddRes( index, typeof( Cloth ), 1044286, 25, 1044287 );
-
-			// Cooking
-			index = AddCraft( typeof( StoneOvenEastDeed ), 1044299, 1044345, 68.4, 93.4, typeof( Log ), 1044041, 85, 1044351 );
-			AddSkill( index, SkillName.Tinkering, 50.0, 55.0 );
-			AddRes( index, typeof( IronIngot ), 1044036, 125, 1044037 );
-			index = AddCraft( typeof( StoneOvenSouthDeed ), 1044299, 1044346, 68.4, 93.4, typeof( Log ), 1044041, 85, 1044351 );
-			AddSkill( index, SkillName.Tinkering, 50.0, 55.0 );
-			AddRes( index, typeof( IronIngot ), 1044036, 125, 1044037 );
-			index = AddCraft( typeof( FlourMillEastDeed ), 1044299, 1044347, 94.7, 119.7, typeof( Log ), 1044041, 100, 1044351 );
-			AddSkill( index, SkillName.Tinkering, 50.0, 55.0 );
-			AddRes( index, typeof( IronIngot ), 1044036, 50, 1044037 );
-			index = AddCraft( typeof( FlourMillSouthDeed ), 1044299, 1044348, 94.7, 119.7, typeof( Log ), 1044041, 100, 1044351 );
-			AddSkill( index, SkillName.Tinkering, 50.0, 55.0 );
-			AddRes( index, typeof( IronIngot ), 1044036, 50, 1044037 );
-			AddCraft( typeof( WaterTroughEastDeed ), 1044299, 1044349, 94.7, 119.7, typeof( Log ), 1044041, 150, 1044351 );
-			AddCraft( typeof( WaterTroughSouthDeed ), 1044299, 1044350, 94.7, 119.7, typeof( Log ), 1044041, 150, 1044351 );
-
-			MarkOption = true;
-			Repair = false;
-
-			SetSubRes( typeof( Log ), 1072643 );
-
-			// Add every material you want the player to be able to choose from
-			// This will override the overridable material	TODO: Verify the required skill amount
-			AddSubRes( typeof( Log ), 1072643, 00.0, 1044041, 1072652 );
-		}
-	}
+    public class DefCarpentry : CraftSystem
+    {
+        public override SkillName MainSkill
+        {
+            get { return SkillName.Carpentry; }
+        }
+
+        public override int GumpTitleNumber
+        {
+            get { return 1044004; } // <CENTER>CARPENTRY MENU</CENTER>
+        }
+
+        private static CraftSystem m_CraftSystem;
+
+        public static CraftSystem CraftSystem
+        {
+            get
+            {
+                if (m_CraftSystem == null)
+                    m_CraftSystem = new DefCarpentry();
+
+                return m_CraftSystem;
```

Comment:
When you see this pattern of:
```csharp
public static CraftSystem CraftSystem
{
    get
    {
        if (m_CraftSystem == null)
            m_CraftSystem = new DefCarpentry();

        return m_CraftSystem;
    }
}
```

It can usually be made simpler with auto-properties and the initialization syntax like this:
```csharp
public static CraftSystem CraftSystem { get; } = new DefCarpentry();
```

If you were using Jetbrains Rider :tm: it would suggest you do it automatically :sunglasses:

Reply:
done